### PR TITLE
Add controllers page, periodically importing from ros2_controllers repo

### PR DESCRIPTION
**Pull request by bmagyar**
I also changed the PR check job to be exactly the same as the build-deploy job but without deploying.

### CODE REVIEW

**Review by destogl — COMMENTED** [review]
IMO we should keep "sphinx-build-check" linter which checks if docs build is possible.

The "sphinx-make-page" pushes the pages to the "gh-pages" branch and if it is used in PRs we will constantly push not accepted docs changes.
This seems to be happening right now when you see "gh-pages" branch...

**Comment by bmagyar** [comment]
@destogl check now, I extended the build job to have a flag for deploying. I think it's important to run the same thing to test build on both workflows.

The contents this PR uses are on https://github.com/ros-controls/ros2_controllers/pull/156

File: .github/workflows/sphinx-build-check.yml [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
+  check-build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - uses: ammaraskar/sphinx-action@master
+    - uses: bmagyar/sphinx-pages@master
       with:
-        docs-folder: "."
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        create_readme: false
```
Comment (destogl) [inline .github/workflows/sphinx-build-check.yml:13]: Should this be automatically `false` when deploy=false  (nice-to-have)

File: .github/workflows/sphinx-build-check.yml [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
-  docs:
+  check-build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - uses: ammaraskar/sphinx-action@master
+    - uses: bmagyar/sphinx-pages@master
       with:
-        docs-folder: "."
+        github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (destogl) [inline .github/workflows/sphinx-build-check.yml:12]: also if there is no deployment, the token is not needed. (nice-to-have)

File: index.rst [hunk 3]
```diff
@@ -7,6 +7,7 @@ Welcome to the ros2_control documentation!
    :hidden:
 
    getting_started.rst
+   ros2_controllers/doc/controllers_index.rst
```
Comment (destogl) [inline index.rst:10]: I propose to  use just `index.rst` file

```suggestion
   ros2_controllers/doc/index.rst
```

**Review by destogl — APPROVED** [review]
Just minor comments and update of the doc-entry point to `index.rst`

File: index.rst (hunk 3 above)
Comment (bmagyar) [reply index.rst:10]: It's much easier to find files when they are not called the same way.

Comment (bmagyar) [reply index.rst:10]: Also note that this isn't necessarily The Entrypoint. It's merely collecting the userdoc files.  A natural extension of this controller-repo inclusion is to run doxygen too, which should probably be a separate page, or perhaps grouped per-controller with the userdoc. We can work that out later.

File: .github/workflows/sphinx-build-check.yml (hunk 1 above)
Comment (bmagyar) [reply .github/workflows/sphinx-build-check.yml]: it is false by default, removed

File: .github/workflows/sphinx-build-check.yml (hunk 2 above)
Comment (bmagyar) [reply .github/workflows/sphinx-build-check.yml]: good point! I had to refactor the gh action code a bit but it works that way now

**Pull request merged**